### PR TITLE
Truncate BPM for large numbers

### LINKED ISSUE #1879: Avoid beatmap data overflowing off the page
https://osu.ppy.sh/beatmapsets/701416#mania/1484517
screen shot 2017-12-03 at 5 09 14 pm

### PR DESCRIPTION
**Pull request by TPGPL**
Intends to resolve #1879

### CODE REVIEW

**Comment by nanaya** [comment]
Probably want to check that this actually solves the problem

mockup by manually changing the one in #4300:

File: resources/assets/coffee/react/_components/beatmap-basic-stats.coffee [hunk 1]
```diff
@@ -32,16 +32,28 @@ formatDuration = (value) ->
   else
     "#{m}:#{_.padStart s, 2, 0}"
 
+truncate = (number) ->
```
Comment (nanaya) [inline resources/assets/coffee/react/_components/beatmap-basic-stats.coffee]: this already exists as `osu.formatNumberSuffixed`

File: resources/assets/coffee/react/_components/beatmap-basic-stats.coffee [hunk 2]
```diff
@@ -32,16 +32,28 @@ formatDuration = (value) ->
...
   div
     className: bn
     for stat in ['total_length', 'bpm', 'count_circles', 'count_sliders']
       value =
         if stat == 'bpm'
-          beatmapset.bpm
+          truncate beatmapset.bpm
         else
-          beatmap[stat]
+          truncate beatmap[stat]
```
Comment (nanaya) [inline resources/assets/coffee/react/_components/beatmap-basic-stats.coffee]: this breaks the following `value` formatting (notably for maps longer than 1000 seconds) .

File: resources/assets/coffee/react/_components/beatmap-basic-stats.coffee [hunk 3]
```diff
@@ -55,6 +67,10 @@ export BeatmapBasicStats = ({beatmapset, beatmap}) ->
         title: osu.trans "beatmapsets.show.stats.#{stat}",
           if stat == 'total_length'
             hit_length: formatDuration(beatmap['hit_length'])
+          else if stat == 'bpm'
+            count: beatmapset.bpm
```
Comment (nanaya) [inline resources/assets/coffee/react/_components/beatmap-basic-stats.coffee]: Store raw/un-suffixed value, remove the conditional above, and format with `formatNumber`.

**Comment by nekodex** [comment]
Hey, so I'm going to close this PR as it doesn't actually fix the issue. In future, please actually test that your fix works before opening a pull request. Thanks.

**Pull request closed without merging**